modules/widgets/top-posts.php

```diff
@@ -510,12 +510,23 @@ function get_posts( $post_ids, $count ) {
 		foreach ( (array) $post_ids as $post_id ) {
 			$post = get_post( $post_id );
 
-			if ( ! $post )
+			if ( ! $post ) {
 				continue;
+			}
+
+			/**
+			 * Attachment pages use the 'inherit' post status by default.
+			 * To be able to remove attachment pages from private and password protect posts,
+			 * we need to replace their post status by the parent post' status.
+			 */
+			if ( 'inherit' == $post->post_status && 'attachment' == $post->post_type ) {
+				$post->post_status = get_post_status( $post_id );
+			}
 
 			// hide private and password protected posts
-			if ( 'publish' != $post->post_status || ! empty( $post->post_password ) || empty( $post->ID ) )
+			if ( 'publish' != $post->post_status || ! empty( $post->post_password ) || empty( $post->ID ) ) {
```

Comment:
Is checking `empty( $post->ID )` necessary? `$post->ID` is the same than `$post_id` and if it was empty then line 513 would be true and it would skip, so I don't think it's necessary to check it again here, unless I'm missing something.